src/program/frame.rs

```diff
@@ -87,3 +126,27 @@ impl FrameSet {
             .collect()
     }
 }
+
+pub(crate) enum FrameMatchCondition<'a> {
+    /// Match all frames in the set
+    All,
+
+    /// Match all frames which shares any one of these names
+    AnyOfNames(&'a [String]),
```

Comment:
Why slices here? You're using vectors below (because of needing to put recursive ones on the heap), and if these get large enough, I imagine a set-based solution would be faster.

Reply:
Removing the lifetime stuff would probably simplify downstream code as well.

Reply:
I suppose it makes sense if you see them all as slices but the Vec as just a `Box<&[_]>` due to the type recursion